spec/nanoc/base/repos/config_loader_spec.rb

```diff
@@ -52,6 +52,25 @@
         expect(subject[:parent_config_file]).to be_nil
       end
     end
+
+    context 'config file present and environment defined' do
+      before do
+        File.write('nanoc.yaml', YAML.dump({ foo: 'bar', environments: { test: { foo: 'test-bar' }, default: { foo: 'default-bar' } } }))
+      end
+
+      it 'returns the configuration' do
+        expect(subject).to be_a(Nanoc::Int::Configuration)
+      end
+
+      it 'has the test environment custom option' do
+        allow(ENV).to receive(:fetch).with('NANOC_ENV', 'default').and_return('test')
+        expect(subject[:foo]).to eq('test-bar')
+      end
+
+      it 'has the default environment custom option' do
+        expect(subject[:foo]).to eq('default-bar')
```

Comment:
Add a test for options defined not within the `default` env, but globally (should return `bar`)